src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -92,6 +84,15 @@ public ScriptHostManager(ScriptHostConfiguration config, IScriptEventManager eve
             _structuredLogWriter = new StructuredLogWriter(EventManager, config.RootLogPath);
             _performanceManager = hostPerformanceManager ?? new HostPerformanceManager(settingsManager, _config.HostHealthMonitor);
 
+            if (config.FileWatchingEnabled && !settingsManager.FileSystemIsReadOnly)
```

Comment:
Isn't the check for `!settingsManager.FileSystemIsReadOnly` redundant here, since you set `FileWatchingEnabled` to false if it's read-only?

Reply:
Unfortunately no - this ctor is called before any hosts are created/initialized. This is the manager of hosts so creates them later at which time it will read and apply host.json settings. A little confusing as is I agree.